domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt

```diff
@@ -230,15 +244,28 @@ class ExplorationProgressController @Inject constructor(
           "Cannot submit an answer while another answer is pending."
         }
         lateinit var hint: Hint
+        val ephemeralState = explorationProgress.stateDeck.getCurrentEphemeralState()
         try {
-          explorationProgress.stateDeck.submitHintRevealed(state, hintIsRevealed, hintIndex)
+          explorationProgress.stateDeck.submitHintRevealed(
+            ephemeralState.state,
+            hintIsRevealed,
+            hintIndex
+          )
           hint = explorationProgress.stateGraph.computeHintForResult(
-            state,
+            ephemeralState.state,
             hintIsRevealed,
             hintIndex
           )
-          explorationProgress.stateDeck.pushStateForHint(state, hintIndex)
+          explorationProgress.stateDeck.pushStateForHint(ephemeralState.state, hintIndex)
         } finally {
+          hintHandler.notifyHintIsRevealed(hintIndex)
+          // Schedule a new hints or solution or show a new hint or solution immediately based on
```

Comment:
```suggestion
          // Schedule, or show immediately, a new hint or solution based on
```

(May require some re-wrapping with the next line).

Reply:
done